### ISSUE TITLE
Build fails for common-config

### ISSUE THREAD

**Issue by anshumandeb** [confluentinc/common#170]
running mvn -pl :common-config package from the parent folder gives the following error:

[ERROR] Failed to execute goal on project common-config: Could not resolve dependencies for project io.confluent:common-config:jar:5.2.0-SNAPSHOT: The following artifacts could not be resolved: io.confluent:common-utils:jar:5.2.0-SNAPSHOT, io.confluent:assembly-plugin-boilerplate:zip:resources:5.2.0-SNAPSHOT: Failure to find io.confluent:common-utils:jar:5.2.0-SNAPSHOT in http://packages.confluent.io/maven/ was cached in the local repository, resolution will not be reattempted until the update interval of confluent has elapsed or updates are forced -> [Help 1]

**Comment by hashhar** [comment]
Any updates on this? This is impeding work on confluentinc/kafka-connect-jdbc#401.

**Comment by RoyalFeng** [comment]
can some one slove this problem？ I have met same issue.

**Comment by cricket007** [comment]
@RoyalFeng It's not clear what specific error you are getting, but you will need to build Kafka source code before you can build this repo from the master branch. 

- Kafka - clone `https://github.com/confluentinc/kafka.git` and build with `./gradlew installAll`
- Common - clone `https://github.com/confluentinc/common` and build with `mvn install`

**Comment by ewencp** [comment]
There's nothing broken here, you are trying to build a single module without building its parent modules, and those SNAPSHOTS aren't going to be published in maven central or Confluent's production release maven repository. You should build the full project. If you are just trying to develop downstream projects, you can `mvn install` from the root of this project.

**Comment by daxiangxiaoyi** [comment]
> * https://github.com/confluentinc/kafka.git

I followed your steps,but still failed.

**Comment by hashhar** [comment]
@daxiangxiaoyi If you can afford to, please clear your local maven repo or at least move it.

```sh
mv ~/.m2 ~/.m2.old
git clone https://github.com/confluentinc/kafka.git
cd kafka
./gradlew installAll
git clone https://github.com/confluentinc/common.git
cd common
mvn install
```

The above works for me everytime.

**Comment by dnascimb** [comment]
This works with `kafka` project on `trunk` and `common` project on `master` branch. If however, I switch the `common` project to branch `5.2.x`, it fails as it is unable to find the snapshots. Are there other conditions that need to be met to make this work?

Here is the failure when running `mvn install` from `common` project when on branch `5.2.x` after successful installation of `kafka` project:
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.691 s
[INFO] Finished at: 2019-03-11T12:37:59-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project common-logging: Could not resolve dependencies for project io.confluent:common-logging:jar:5.2.0-SNAPSHOT: The following artifacts could not be resolved: org.apache.kafka:connect-api:jar:2.2.0-SNAPSHOT, org.apache.kafka:connect-json:jar:2.2.0-SNAPSHOT: Failure to find org.apache.kafka:connect-api:jar:2.2.0-SNAPSHOT in http://packages.confluent.io/maven/ was cached in the local repository, resolution will not be reattempted until the update interval of confluent has elapsed or updates are forced -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :common-logging
```

Do I need to switch the `kafka` project to an alternate branch, and if so, how do I determine what branch is compatible with the branch I have checked out from `common`?

**Comment by ewencp** [comment]
You need to use the corresponding Kafka branch for the release. The mapping is documented [here](https://docs.confluent.io/current/installation/versions-interoperability.html#cp-and-apache-kafka-compatibility). So for example if you want to build the 5.1.x branch you would build the 2.1 branch of Kafka.

**Comment by cricket007** [comment]
Also, note, the build reason for the failure would be above the `[INFO] BUILD FAILURE` line. Without that information, it's difficult to tell what the solution could be.

**Comment by dnascimb** [comment]
Thanks @ewencp and @cricket007 👍

**Comment by imranrajjad** [comment]
so what kafka version is required for 5.3.0-SNAPSHOT?

**Comment by ewencp** [comment]
2.3-SNAPSHOT. You can find the kafka version, regardless of branch, here https://github.com/confluentinc/common/blob/master/pom.xml#L62 (and it'll also be reported in the error message if you try to build without having the dependency installed).

**Comment by cricket007** [comment]
And you can find the version being built of the Kafka repo here - https://github.com/confluentinc/kafka/blob/trunk/gradle.properties#L23

**Comment by MohammadChalaki** [comment]
> @daxiangxiaoyi If you can afford to, please clear your local maven repo or at least move it.
> 
> ```shell
> mv ~/.m2 ~/.m2.old
> git clone https://github.com/confluentinc/kafka.git
> cd kafka
> ./gradlew installAll
> git clone https://github.com/confluentinc/common.git
> cd common
> mvn install
> ```
> 
> The above works for me everytime.

Using ./gradlew installAll I get the error:
Task 'installAll' not found in root project 'kafka'. Some candidates are: 'install'.

**Comment by OneCricketeer** [comment]
Try reading the README

https://github.com/confluentinc/kafka#installing-the-jars-to-the-local-maven-repository

**Comment by MohammadChalaki** [comment]
> Try reading the README
> 
> https://github.com/confluentinc/kafka#installing-the-jars-to-the-local-maven-repository

I've done everything as supposed, still getting error
The POM for io.confluent:resolver-maven-plugin:jar:0.4.0 is missing, no dependency information available

**Comment by OneCricketeer** [comment]
Error suggests you'll need to install this now

https://github.com/confluentinc/resolver-maven-plugin

**Comment by ewencp** [comment]
That's correct, this is a very recent change in requirements. I think we're considering just pushing that plugin to maven central so it's easier to get to.

**Comment by MohammadChalaki** [comment]
> That's correct, this is a very recent change in requirements. I think we're considering just pushing that plugin to maven central so it's easier to get to.

It will be great, since https://github.com/confluentinc/resolver-maven-plugin is for 5 years ago and trying it didnt help with the issue.

**Comment by OneCricketeer** [comment]
The version you're getting an error for is 9 days old

**Comment by jbenoit2011** [comment]
I encounter troubles too to install JAR in my local Maven:

1. I cloned `confluentinc/resolver-maven-plugin`, checkout tag `0.4.0`, ran `mvn clean install` and it succeed
2. I cloned `confluentinc/kafka` repo, checkout branch `v6.1.0-beta200907192126` and ran `.gradlewAll install` and it succeed
3. I cloned `confluentinc/common` repo, checkout branch `v6.1.0-beta200907192126`, ran `mvn clean install` and it failed with:
```
[ERROR] Failed to execute goal io.confluent:resolver-maven-plugin:0.4.0:resolve-kafka-range (default) on project common-parent: : MojoExecutionException: No matching ce-kafka version found for constraint: 'org.apache.kafka:kafka-clients::[6.1.0-0, 6.1.1-0), excluding snapshots'. -> [Help 1]
```

What's wrong?
Is there somewhere a complete and up-to-date manual about how to build confluent projects? My goal is to build latest schema-registry version to try a patch not yet published on Confluent Maven.

Thanks.

**Comment by OneCricketeer** [comment]
If you want to "build latest", for active development, then use master branches

**Comment by ldkhanh** [comment]
Add this into your pom file

  ```
<repositories>
        <repository>
            <id>confluent</id>
            <url>https://packages.confluent.io/maven/</url>
        </repository>
 </repositories>
```

**Comment by raphaelauv** [comment]
also you can just checkout a release version

like 

```shell
git checkout v6.0.1
```
